Add 3D tasks

dylanebert opened the pull request:
Adding `text-to-3D` and `image-to-3D` tasks.

The output could be
- A mesh file. This could be `.obj`, `.gltf`, or `.glb`. OBJ and GLTF formats, by convention, have other files in the same directory with the same name. GLB is fully self-contained.
- A splat file. This could be `.ply` or `.splat`. PLY is the large, lossless format introduced with Gaussian Splatting, while `.splat` is an emerging compression format.

Current Models
- [shap-e](https://huggingface.co/openai/shap-e)
- [zero123](https://huggingface.co/bennyguo/zero123-xl-diffusers)

Demos
- [shap-e](https://huggingface.co/spaces/hysts/Shap-E)
- [One-2-3-45](https://huggingface.co/spaces/One-2-3-45/One-2-3-45)

osanseviero commented on packages/tasks/src/pipelines.ts:
```diff
@@ -635,6 +635,16 @@ export const PIPELINE_DATA = {
 		modality: "cv",
 		color: "yellow",
 	},
+	"text-to-3d": {
```
Feel free to add `subtasks` for different outputs if you think it's worth it, but no need for now

osanseviero replied:
We should start tagging some repos with this, otherwise https://huggingface.co/datasets?other=text-to-3d will be empty. You can add `hideInDatasets: false` here and it will not show the tag at left of hf.co/datasets, but I think we do have 3D datasets to add this tag.

osanseviero replied:
We need to tag https://huggingface.co/bennyguo/zero123-xl-diffusers with this type, feel free to add a PR! We should similarly start adding PRs to add models to https://huggingface.co/models?other=image-to-3d

osanseviero reviewed (commented):
Very exciting! 🔥 There are a couple of pieces that will help tie all together

* Please also add it to https://main.com/huggingface/huggingface.js/blob/41ddbb0ba07ee0c1eff74e5507203937af02109d/packages/hub/src/types/public.d.ts#L46
* You can also choose an icon (at carbon.sh) for the tasks, and add it to https://github.com/huggingface/huggingface.js/blob/main/packages/widgets/src/lib/components/PipelineIcon/PipelineIcon.svelte#L46 cc @gary149 @Reeray 

cc @gary149 for review

dylanebert commented:
> Very exciting! 🔥 There are a couple of pieces that will help tie all together
> 
> * Please also add it to https://main.com/huggingface/huggingface.js/blob/41ddbb0ba07ee0c1eff74e5507203937af02109d/packages/hub/src/types/public.d.ts#L46
> * You can also choose an icon (at carbon.sh) for the tasks, and add it to https://github.com/huggingface/huggingface.js/blob/main/packages/widgets/src/lib/components/PipelineIcon/PipelineIcon.svelte#L46 cc @gary149 @Reeray
> 
> cc @gary149 for review

Thanks! I have added to public.d.ts and added icons. The formatting from https://icones.js.org/collection/carbon didn't exactly match that in the existing icons, so I manually adjusted them, but haven't been able to confirm that they're correct.

osanseviero reviewed (approved):
🔥

pcuenca commented on packages/tasks/src/tasks/image-to-3d/about.md:
Are these empty files necessary? (just curious)

dylanebert replied:
Will go ahead and remove them

pcuenca reviewed (approved):
Exciting!

merveenoyan commented on packages/tasks/src/tasks/index.ts:
```diff
@@ -161,6 +163,8 @@ export const TASKS_DATA: Record<PipelineType, TaskData | undefined> = {
 	"zero-shot-classification": getData("zero-shot-classification", zeroShotClassification),
 	"zero-shot-image-classification": getData("zero-shot-image-classification", zeroShotImageClassification),
 	"zero-shot-object-detection": getData("zero-shot-object-detection", placeholder),
+	"text-to-3d": undefined,
```
You can put placeholder there instead so that people see call to contribution

merveenoyan commented on packages/tasks/src/tasks/index.ts:
```diff
@@ -161,6 +163,8 @@ export const TASKS_DATA: Record<PipelineType, TaskData | undefined> = {
 	"zero-shot-image-classification": getData("zero-shot-image-classification", zeroShotImageClassification),
 	"zero-shot-object-detection": getData("zero-shot-object-detection", placeholder),
+	"text-to-3d": undefined,
+	"image-to-3d": undefined,
```
You can also remove `about.md` and `data.ts` files.

merveenoyan reviewed (commented):
Thanks!

dylanebert commented:
Thanks @merveenoyan @osanseviero @pcuenca @coyotte508 

I have removed the extra doc files and set the placeholders
